core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala

```diff
@@ -162,9 +163,15 @@ abstract class RpcEnvSuite extends SparkFunSuite with BeforeAndAfterAll {
     val rpcEndpointRef = anotherEnv.setupEndpointRef("local", env.address, "ask-timeout")
     try {
       val e = intercept[Exception] {
-        rpcEndpointRef.askWithRetry[String]("hello", 1 millis)
+        rpcEndpointRef.askWithRetry[String]("hello", new RpcTimeout(1 millis, shortProp))
       }
       assert(e.isInstanceOf[TimeoutException] || e.getCause.isInstanceOf[TimeoutException])
+      e match {
+        case te: TimeoutException =>
+          assert(te.getMessage().contains(shortProp))
+        case e: Exception =>
+          assert(e.getCause().getMessage().contains(shortProp))
```

Comment:
Do you know how we end up with a non-`TimeoutException` here?
(I realize that before there was already a check for that, you are just keeping what was there -- just curious if you know)